### DISCUSSION TITLE
Question About Handling the Reception of STOP_SENDING frame

### DISCUSSION THREAD

**Discussion by LiangLouise** [ngtcp2/ngtcp2#1148] [Q&A] [answered]
I wonder if there is any way to know the event of receiving a STOP_SENDING frame and retrieve the app_error_code.

When receiving a STOP_SENDING frame, `conn_recv_stop_sending` would do a lot of internal jobs, however, it doesn't have any callbacks to tell the user that some stream's sending part got shut down and provide the app_error_code.

By reading the doc and code, it seems the only way to know if a stream's sending part is shut down is by calling `ngtcp2_conn_writev_stream()` and getting the error value `NGTCP2_ERR_STREAM_SHUT_WR`. But I cannot get what the app_error_code is in STOP_SENDING in this way since it won't be provided to the user until the `stream_close` callback is called.

Did I miss anything?

**Comment by tatsuhiro-t** [comment, marked as answer]
That is the correct explanation of the current state of the art.

**Reply by LiangLouise** [reply]
Thank you for the quick reply and confirmation!

Then I guess the easiest way for me would be adding a callback to be called in `conn_recv_stop_sending` so it doesn't have to wait until calling `ngtcp2_conn_writev_stream` next time and the user code can also get back the app_error_code meanwhile.
